game/PsyDoom/Game.cpp

```diff
@@ -232,6 +232,12 @@ void getUserGameSettings(GameSettings& settings) noexcept {
     settings.bFixSpriteVerticalWarp         = Config::gbFixSpriteVerticalWarp;
     settings.bAllowMultiMapPickup           = Config::gbAllowMultiMapPickup;
     settings.bEnableMapPatches_GamePlay     = Config::gbEnableMapPatches_GamePlay;
+    settings.bNoFriendlyFire                = Config::gbNoFriendlyFire;
+    settings.fragLimit                      = Config::gFragLimit;
+    settings.bExitDisabled                  = Config::gbExitDisabled;
+    settings.preserveAmmoFactor             = Config::gPreserveAmmoFactor;
+    settings.bPreserveKeys                  = Config::gbPreserveKeys;
+    settings.bMPThings                      = Config::gbMPThings;
 
```

Comment:
Please set these values explicitly also in `getClassicDemoGameSettings`. That way it's clearly marked what values these settings should have to replicate the behavior of the original PSX Doom.

Reply:
Please also add corresponding lines of code for these new fields to `GameSettings::byteSwap()`. I know they are one byte now and the swap is a no-op, but explicitly swapping in that function helps verify that no fields were missed (for swapping) and guards against future changes of the underlying type.

Reply:
3f175651b823c8b4b4022729a2309a2873adafc2